java/memory/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java

```diff
@@ -95,4 +143,26 @@ public static long getByteBufferAddress(ByteBuffer buf) {
 
   private MemoryUtil() {
   }
+
+  /**
+   * Create nio byte buffer.
+   */
+  public static ByteBuffer directBuffer(long address, int capacity) {
+    if (DIRECT_BUFFER_CONSTRUCTOR != null) {
+      if (capacity < 0) {
+        throw new IllegalArgumentException("Capacity is negative, has to be positive or 0");
+      }
+      try {
+        return (ByteBuffer) DIRECT_BUFFER_CONSTRUCTOR.newInstance(address, capacity);
+      } catch (Throwable cause) {
+        // Not expected to ever throw!
+        if (cause instanceof Error) {
+          throw (Error) cause;
```

Comment:
Is there a special reason for the cast here?

Reply:
None, only that it was how the netty developers wrote it. I thought I would copy their lead. However, I have simplified it now,